### PR TITLE
 convert dashboard to new scope/store pattern

### PR DESCRIPTION
**Pull request by bluwave**

### CODE REVIEW

File: Core/Core/Use Cases/GetCourses.swift [hunk 1]
```diff
@@ -16,33 +16,34 @@
...
-        let ttl = TTLOperation(key: "get-courses", database: env.database, operation: paginated, force: force)
-        super.init(operations: [ttl])
+    public func write(response: [APICourse]?, urlResponse: URLResponse?, to client: PersistenceClient) throws {
+        guard let response = response else {
+            return
+        }
+
+        for item in response {
+            Course.save(item, in: client)
+        }
```
Comment (vanwagonet) [inline Core/Core/Use Cases/GetCourses.swift]: Could `Course` be a `WriteableModel`, and this write method not need to be defined explicitly?

Comment (bluwave) [reply Core/Core/Use Cases/GetCourses.swift]: fixed

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-12069](https://instructure.atlassian.net/browse/MBL-12069)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 90.43% | 90.61% | -0.18%
**Student** | 94.52% | 94.52% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.3% | 92.3% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by vanwagonet** [comment]
This appears to be working, but it doesn't show all pages of data in the CourseList or Dashboard. Should we try to get that done right now? or defer since this dashboard isn't used in the actual app yet?

**Comment by vanwagonet** [comment]
I'm also seeing build errors when trying to build the `Canvas` scheme.

**Comment by bluwave** [comment]
@vanwagonet i checked what is being shown now and before this pr.  The data appears to be the same.  Since this is just trying to get arch design stuff swapped out, i say we leave as is and fix at another time.  There is no apparent change from the dashboard previously which apparently is wrong.  The All Courses list was just displaying all courses for both current and past enrollments.  I changed it to show all for current and favorites for past.  This too was wrong previous to this pr.  I say we lave as is.  Address in another ticket.

**Comment by bluwave** [comment]
> I'm also seeing build errors when trying to build the `Canvas` scheme.

this should be fixed

**Pull request merged**